src/platform-implementation-js/dom-driver/gmail/widgets/gmail-mole-view-driver.ts

```diff
@@ -28,7 +31,123 @@ export type MoleOptions = {
 
 const INBOXSDK_CLASS = 'inboxsdk__mole_view' as const;
 
+const enum Selector {
+  MoleParent = '.dw .nH > .nH > .no',
+  /**
+   * Compose and SDK mole selector.
+   *
+   * @note 2023-10-11 on :not selectors:
+   *
+   * style*="order: 2147483647;" is the left mole spacer.
+   *
+   * style*="order: 0;" is the right mole spacer.
+   *
+   * .aJl is the chat placeholder.
+   */
+  Mole = `${Selector.MoleParent} > *:not([style*="order: 0;"], [style*="order: 2147483647;"], .aJl)`,
+  ComposeMole = `${Selector.Mole}.nn.nH`,
+  ChatMolePlaceholder = `${Selector.MoleParent} > .aJl`,
+}
+
+const enum Tag {
+  Mole = 'mole',
+  MoleParent = 'moleParent',
+}
+
 class GmailMoleViewDriver {
+  static #page: PageParserTree;
+
+  static {
+    this.#page = new PageParserTree(document, {
+      logError(err, el) {
+        const details = {
+          el,
+          html: el ? censorHTMLtree(el) : null,
+        };
+        // ignore 'errors' that seem to be warnings instead...
+        if (
+          err.message.includes('found element missed by watcher') ||
+          err.message.includes('watcher found element already found by finder')
+        ) {
```

Comment:
we shouldn't ignore these errors, these errors are telling you your watcher and finder are not setting up properly